src/components/jobs_view.tsx

```diff
@@ -18,17 +18,42 @@ const JobList: React.FC<JobListProps> = ({ dbName }) => {
   const db = new PouchDB(dbName);
   const [sortedJobs, setSortedJobs] = useState<any[]>([]);
   const [jobsList, setJobsList] = useState<any[]>([]);
+  const [isAddModalOpen, setIsAddModalOpen] = useState(false);
+  const [modalOpenMap, setModalOpenMap] = useState<{ [jobId: string]: boolean }>({});
+
+
+  const openAddModal = () => {
+    setIsAddModalOpen(true);
+  };
+
+  const closeAddModal = () => {
+    setIsAddModalOpen(false);
+  };
+
+  const validateInput = [
+    {
+      validator: (input:string) => {
+        // Restrict the character set to [a-zA-Z0-9-_#:>]
+        const regex = /^(?![\-_#:>])[a-zA-Z0-9\-_#:>]+$/;
+        return regex.test(input);
+      },
+      errorMsg: 'Invalid characters used. Please use only [a-zA-Z0-9-_#:>] and do not start with a speical character.',
```

Comment:
Since our users won't understand regular expressions, we'll need to write it out: "The name must be no more than 64 characters starting with a letter, a number, or # and followed by letters, numbers, #, dash, comma, and single spaces between other characters."

Notice that I've suggested some changes to the constraints.